src/VisualStudio/CSharp/Impl/Options/AdvancedOptionPageControl.xaml.cs

```diff
@@ -172,16 +174,33 @@ private void UpdatePullDiagnosticsOptions()
             }
         }
 
-        private void Enable_pull_diagnostics_experimental_requires_restart_Checked(object sender, RoutedEventArgs e)
+        private void Enable_pull_diagnostics_experimental_requires_restart_CheckedChanged(object sender, RoutedEventArgs e)
         {
-            this.OptionStore.SetOption(InternalDiagnosticsOptions.NormalDiagnosticMode, DiagnosticMode.Pull);
-            UpdatePullDiagnosticsOptions();
-        }
+            // Three state is only valid for the initial option state (default).  If changed we only
+            // allow the checkbox to be on or off.
+            Enable_pull_diagnostics_experimental_requires_restart.IsThreeState = false;
```

Comment:
This is exactly what we do for other feature flagged checkboxes, e.g. https://github.com/dotnet/roslyn/blob/315c2e149ba7889b0937d872274c33fcbfe9af5f/src/VisualStudio/CSharp/Impl/Options/IntelliSenseOptionPageControl.xaml.cs#L64